logictest: add 3node-tenant configuration

tbg opened the pull request:
In this configuration, the `*gosql.DB`s used by the test are started
against a SQL tenant server pointed at one of the TestServers.

The goal is to make this a default config over time that skips only
those test files which are known to test functionality not available
in a multi-tenancy setup.

For now, it's exercised by a newly added logic test that tests some
of the expected errors.

cc @nvanbenschoten @andreimatei 

Release note: None

tbg committed:
server: use real RuntimeSampler in test tenant server

Release note: None

tbg committed:
server: disable protected timestamps for SQL tenants

Tenants aren't going to be able to use protected timestamps because the
KV backend won't live up to its end of the contract. We'll need
engineering work to make it happen.

As far as I know, `IMPORT INTO` is the only operation that uses
protected timestamps for correctness. Other uses can selectively ignore
the result of the `Protect()` operation, though this change is not
made here.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/47755)
<!-- Reviewable:end -->

asubiotto reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> Reviewed 2 of 2 files at r1, 1 of 1 files at r2, 1 of 2 files at r3.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/47755#-:-M5WEjKuEBZ79BG9NE_x:bg7igkv)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @asubiotto and @tbg)

___
*[pkg/sql/logictest/testdata/logic_test/tenant_unsupported, line 14 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/47755#-M5WES5B-XNuaHYsHIkl:-M5WES5B-XNuaHYsHIkm:b-azhq9j) ([raw file](https://github.com/cockroachdb/cockroach/blob/371e5efe5b07ca8618ae395f010b4e51b305095c/pkg/sql/logictest/testdata/logic_test/tenant_unsupported#L14)):*
> ```
> # This shouldn't work if it uses DistSQL.
> # TODO(tbg): figure out why it works. We do still give a DistSender to SQL, but
> # it's unclear why that's enough - the status server and node liveness don't work.
> ```

If there are problems during physical planning related to figuring out where to schedule table readers, we try to fall back to the gateway as much as possible. This is probably what's going on here.


<!-- Sent from Reviewable.io -->

tbg committed:
logictest: add 3node-tenant configuration

In this configuration, the `*gosql.DB`s used by the test are started
against a SQL tenant server pointed at one of the TestServers.

The goal is to make this a default config over time that skips only
those test files which are known to test functionality not available
in a multi-tenancy setup.

For now, it's exercised by a newly added logic test that tests some
of the expected errors.

Release note: None

tbg commented:
Fixed lint error

TFTR! I updated the comment in the test. I also filed an issue about disabling it properly: https://github.com/cockroachdb/cockroach/issues/47902

tbg commented:
bors r=asubiotto
